lib/modules/manager/maven/extract.ts

```diff
@@ -76,6 +78,75 @@ function containsPlaceholder(str: string | null | undefined): boolean {
   return !!str && regEx(/\${[^}]*?}/).test(str);
 }
 
+function getCNBDependencies(
+  nodes: XmlElement[],
+  config: ExtractConfig,
+): PackageDependency[] {
+  const deps: PackageDependency[] = [];
+  nodes.forEach((node) => {
+    const depString = node.val.trim();
+    if (isDockerRef(depString)) {
+      const dep = getDockerDep(
+        depString.replace(DOCKER_PREFIX, ''),
+        true,
+        config.registryAliases,
+      );
+      dep.registryUrls = [];
+      dep.fileReplacePosition = node.position;
+      if (dep.currentValue || dep.currentDigest) {
+        deps.push(dep);
+      }
+    }
+  });
+  return deps;
+}
+
+function getAllCNBDependencies(
+  node: XmlDocument,
+  config: ExtractConfig,
+): PackageDependency[] | null {
+  const pluginNodes =
+    node.childNamed('build')?.childNamed('plugins')?.childrenNamed('plugin') ??
+    [];
+  pluginNodes.filter((pluginNode) => {
+    return (
+      pluginNode.valueWithPath('groupId')?.trim() ===
+        'org.springframework.boot' &&
+      pluginNode.valueWithPath('artifactId')?.trim() ===
+        'spring-boot-maven-plugin'
+    );
+  });
+  if (pluginNodes.length !== 1) {
+    return null;
+  }
```

Comment:
If you only handle the case where there's one result, use `.find()` instead of `.filter()`.

Reply:
Done